sql-plugin/src/main/scala/com/nvidia/spark/rapids/PrioritySemaphore.scala

```diff
@@ -110,12 +111,16 @@ class PrioritySemaphore[T](val maxPermits: Long)
 
   private def commitAcquire(numPermits: Long): Unit = {
     occupiedSlots += numPermits
+    currentConcurrentGpuTasksNum += 1
+    // Report current concurrent tasks to GpuTaskMetrics, let it handle max tracking
+    GpuTaskMetrics.get.recordConcurrentGpuTasks(currentConcurrentGpuTasksNum)
```

Comment:
```suggestion
    GpuTaskMetrics.foreach(_.recordConcurrentGpuTasks(currentConcurrentGpuTasksNum))
```
just to be safe.

Reply:
GpuTaskMetrics is not a Option here, so it does not have foreach method